tests/test_routes_agent_versions.py

```
@@ -0,0 +1,527 @@
+    async def _fake(container, cmd, timeout=60):
+        if cmd[0] == "git" and cmd[1] == "-C" and cmd[2] == "/root":
+            git_args = cmd[3:]
+            result = subprocess.run(
+                ["git", "-C", str(fixture_repo), *git_args],
+                capture_output=True,
+                text=True,
+            )
+            return result.returncode, result.stdout
+        if cmd[0] == "bash" and cmd[1] == "-c":
+            script = cmd[2].replace("/root", str(fixture_repo))
+            result = subprocess.run(
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)